core/services/helper/main.py

```diff
@@ -317,7 +317,7 @@ def detect_service(port: int) -> ServiceInfo:
             "127.0.0.1", port=port, path="/", timeout=1.0, method="GET", follow_redirects=10
         )
         log_msg = f"Detecting service at port {port}"
-        if response.status != http.client.OK:
+        if response.status == http.client.BAD_REQUEST:
```

Comment:
could you explain the reason behind this?

Reply:
shouldn't the servers give a 200 when valid?

Reply:
In the commit body:
> Sometimes it can return a redirect or any other, register service will take care for us

Reply:
My case was moving openvscode to vscode where the / was not an OK but something

Reply:
okay, it looks prone to false positives, but it might be better than false negatives.